packages/core-mobile/app/new/features/accountSettings/components/AcccountList.tsx

```diff
@@ -26,18 +27,25 @@ export const AccountList = (): React.JSX.Element => {
   } = useTheme()
   const dispatch = useDispatch()
   const { navigate } = useRouter()
+
   const activeAccount = useSelector(selectActiveAccount)
   const accountCollection = useSelector(selectAccounts)
   const flatListRef = useRef<FlatList>(null)
 
-  const accounts = useMemo(
-    () => Object.values(accountCollection),
-    [accountCollection]
-  )
+  const { recentAccountIndexes, addRecentAccount } = useRecentAccounts()
 
-  const accountsToDisplay = useMemo(() => {
-    return accounts.slice(0, 5)
-  }, [accounts])
+  const recentAccounts = useMemo(() => {
+    return recentAccountIndexes.map(index => accountCollection[index])
+  }, [accountCollection, recentAccountIndexes])
+
+  useEffect(() => {
+    if (activeAccount?.index != null) {
+      addRecentAccount(activeAccount.index)
```

Comment:
I don't think we should do this here since this will only trigger when this component is mounted and we do change active account in other places as well (in some background logic for example). you should do this in a listener in account slice and listen to setActiveAccountIndex action instead.